multimodal/src/autogluon/multimodal/utils/metric.py

```diff
@@ -75,61 +81,60 @@ def infer_metrics(
+                f"Metric {eval_metric_name} and metric {validation_metric_name} are used "
+                f"as the evaluation metric and the validation metric for {problem_type} in matching tasks by default. "
+            )
+            return validation_metric_name, eval_metric_name
+        else:
+            raise NotImplementedError(f"Problem type: {problem_type} is not yet supported for matching!")
+
+    if eval_metric_name is not None:
+        if eval_metric_name.lower() not in EVALUATION_METRICS[problem_type]:
+            warnings.warn(
+                f"Metric {eval_metric_name} is not supported as the evaluation metric for {problem_type}. "
+                f"The evaluation metric is changed to {EVALUATION_METRICS_FALLBACK[problem_type]} by default."
+            )
+            eval_metric_name = EVALUATION_METRICS_FALLBACK[problem_type]
+
+        if eval_metric_name.lower() in VALIDATION_METRICS[problem_type]:
+            logger.info(f"Metric {eval_metric_name} is used as the validation metric. ")
+            validation_metric_name = eval_metric_name
+        else:
+            validation_metric_name = VALIDATION_METRICS_FALLBACK[problem_type]
 
         warnings.warn(
             f"Currently, we cannot convert the metric: {eval_metric_name} to a metric supported in torchmetrics. "
             f"Thus, we will fall-back to use accuracy for multi-class classification problems "
             f", ROC-AUC for binary classification problem, and RMSE for regression problems.",
             UserWarning,
         )
-
-    if problem_type == MULTICLASS:
-        if is_matching:
-            eval_metric_name = SPEARMANR
-        else:
-            eval_metric_name = ACCURACY
-    elif problem_type == NER:
-        return NER_TOKEN_F1, OVERALL_F1
-    elif problem_type == BINARY:
-        eval_metric_name = ROC_AUC
-    elif problem_type == REGRESSION:
-        if is_matching:
-            eval_metric_name = SPEARMANR
-        else:
-            eval_metric_name = RMSE
-    elif problem_type == FEW_SHOT_CLASSIFICATION:
-        eval_metric_name = ACCURACY
-    elif problem_type in [OBJECT_DETECTION, OPEN_VOCABULARY_OBJECT_DETECTION]:
-        if (not validation_metric_name) or validation_metric_name.lower() == MAP:
-            return MAP, MAP
-        elif validation_metric_name.lower() == DIRECT_LOSS:
-            return DIRECT_LOSS, MAP
-        else:
-            raise ValueError(
-                f"Problem type: {problem_type}, validation_metric_name: {validation_metric_name} is not supported!"
-            )
-    elif problem_type is None and is_matching:
-        return RECALL, NDCG
     else:
-        raise NotImplementedError(f"Problem type: {problem_type} is not supported yet!")
-
-    validation_metric_name = eval_metric_name
+        eval_metric_name = EVALUATION_METRICS_FALLBACK[problem_type]
+        validation_metric_name = VALIDATION_METRICS_FALLBACK[problem_type]
 
     return validation_metric_name, eval_metric_name
```

Comment:
The current logic ignores the `validation_metric_name` from `infer_metrics`'s input. Do we still support users to customize validation metric? I think previously it's mainly used by detection for the map issue.

Reply:
We can ignore for now. But we might need to support this later.